moveit_setup_assistant/src/tools/moveit_config_data.cpp

```diff
@@ -529,6 +529,29 @@ bool MoveItConfigData::outputFakeControllersYAML(const std::string& file_path)
     emitter << YAML::EndSeq;
     emitter << YAML::EndMap;
   }
+
+  emitter << YAML::EndSeq;
+
+  // Add an initial pose for each group
+  emitter << YAML::Key << "initial";
+  emitter << YAML::Value << YAML::BeginSeq;
+
+  for (srdf::Model::Group& group : srdf_->groups_)
+  {
+    const robot_model::JointModelGroup* joint_model_group = getRobotModel()->getJointModelGroup(group.name_);
+    emitter << YAML::BeginMap;
+    const std::vector<const robot_model::JointModel*>& joint_models = joint_model_group->getActiveJointModels();
+    emitter << YAML::Key << "group";
+    emitter << YAML::Value << group.name_;
+
+    emitter << YAML::Key << "pose";
+    if (!srdf_->group_states_.empty())
+      emitter << YAML::Value << srdf_->group_states_[0].name_;
+    else
+      emitter << YAML::Value << "home";  // Default entry
```

Comment:
If no group state is available, nothing should be written here. `home` is probably not defined...

Reply:
It isn't, but it also doesn't cause any errors. I left it in as an example, for users to find if they read through the files, so that the option is visible even if they have not defined a group state yet. Feature discovery and that.

I'm not opposed if the consensus is to remove it, but I figured it doesn't hurt.

Reply:
According to the following code, an unknown pose should trigger a warning:
https://github.com/ros-planning/moveit/blob/6b56cd68d76ae29d1ceb255d06be35543dd3f830/moveit_plugins/moveit_fake_controller_manager/src/moveit_fake_controller_manager.cpp#L164-L169

If it doesn't, there is a bug somewhere else.
For feature discovery, I suggest commenting the corresponding pose line in yaml. Not sure though, how to do this programmatically.

Reply:
True, it might throw a warning, and it's not nice to be flooded with those by default :)

I will change it and add a comment instead. Apparently [YAML::Comment works](https://github.com/jbeder/yaml-cpp/wiki/How-To-Emit-YAML).